pyproject.toml

```diff
@@ -1,6 +1,66 @@
+[project]
+name = "stactools-nisar-sim"
```

Comment:
There is quite a lot of changes in terms of how we install dependencies and manage environments in this PR: `pyproject.toml`, `requirements-dev.txt`, the `Dockerfile`, `setup.cfg`. I remember indeed struggling with environment set up when I worked on V1. 

Could you provide a brief explanation of these changes in the PR description ?

Reply:
(I didn't want to necessarily add these changes to the changelog, since the package was never released)